dev/snippets/lib/snippets.dart

```diff
@@ -226,6 +228,13 @@ class SnippetGenerator {
         final File outputFile = output ?? getOutputFile(id);
         stderr.writeln('Writing to ${outputFile.absolute.path}');
         outputFile.writeAsStringSync(app);
+
+        final File metadataFile = File('${outputFile.path}.json');
+        stderr.writeln('Writing metadata to ${metadataFile.absolute.path}');
+        final Map<String, Object> metadata = <String, Object>{
+          'file': path.basename(outputFile.path),
+        };
+        metadataFile.writeAsStringSync(jsonEncoder.convert(metadata));
```

Comment:
I was thinking that you probably actually want to write it in the main.dart, since this class doesn't have all of the metadata available. But, if you want to write it here (which makes sense, other than the metadata not being available), you could pass in a Map with the data in it from the main.

Reply:
I don't mind moving it - I couldn't see what metadata we wanted so didn't know where it existed. I'll move it to `main.dart` and add the metadata I can see (package, library, element names), but if there should be a range in there too, let me know where I can find (or calculate) it.

Reply:
Actually, I'll leave it inside for now simple because the filename may be passed in as null and generated itself, and I'd rather not duplicate that logic outside to keep the filenames the same (should it ever change), and I'll pass in a map (though lmk if you'd prefer the opposite).